torch/csrc/distributed/c10d/default_comm_hooks.cpp

```diff
@@ -6,11 +6,22 @@
 #include <c10d/comm.hpp>
 #include <torch/torch.h>
 
-namespace c10d {
+namespace {
+void convert_complex_tensors(std::vector<at::Tensor> &tensors) {
+  for (auto &tensor : tensors) {
+    if (tensor.is_complex()) {
+      tensor = at::view_as_real(tensor);
+    }
+  }
+}
+
+} // namespace
 
+namespace c10d {
 c10::intrusive_ptr<c10::ivalue::Future> AllReduceCommHook::runHook(
     GradBucket& bucket) {
   std::vector<at::Tensor> tensors = {bucket.getBufferRef()};
+  convert_complex_tensors(tensors);
```

Comment:
I wonder if calling this routine for grad buckets that contain a large number of tensors might cause some regression in production workloads. Could we run internal QPS benchmarks for RoBERTa and ResNet to verify the performance implications here?